base/repl/LineEdit.jl

```diff
@@ -1329,30 +1356,60 @@ function bracketed_paste(s)
     return replace(input, '\t', " "^tabwidth)
 end
 
+beginofline(buf, pos=position(buf)) = findprev(buf.data, '\n' % UInt8, pos)
+
+function endofline(buf, pos=position(buf))
+    eol = findnext(buf.data[pos+1:buf.size], '\n' % UInt8, 1)
+    eol == 0 ? buf.size : pos + eol - 1
+end
+
+# jump_spaces: if cursor is on a ' ', move it to the first non-' ' char on the right
+# if `delete_trailing`, ignore trailing ' ' by deleting them
+function edit_tab(s, jump_spaces=false, delete_trailing=jump_spaces)
+    # Yes, we are ignoring the possiblity
+    # the we could be in the middle of a multi-byte
+    # sequence, here but that's ok, since any
+    # whitespace we're interested in is only one byte
+    buf = buffer(s)
+    i = position(buf)
+    if i != 0 &&
+        let c = buf.data[i]
+            c == UInt8('\n') || c == UInt8('\t') ||
```

Comment:
Actually I got the indentation of the lines below wrong (they should be aligned with the line above). If you think it's more clear, I can put the `let` on the right side of `&&`.